api/src/main/java/org/eclipse/microprofile/openapi/annotations/media/Schema.java

```diff
@@ -399,4 +399,31 @@
      * @since 2.0
      */
     SchemaProperty[] properties() default {};
+
+    /**
+     * Provides a Java class as implementation for additional properties that may be present in instances of this
+     * schema.
+     * 
+     * <p>
+     * If no additional properties are allowed, the value of this property should be set to
+     * {@link org.eclipse.microprofile.openapi.models.media.Schema.FalseSchema FalseSchema.class} which will be rendered
+     * as boolean <code>false</code> in the resulting OpenAPI document.
+     * 
+     * <p>
+     * The default value {@link org.eclipse.microprofile.openapi.models.media.Schema.TrueSchema TrueSchema.class}
+     * indicates that additional properties are allowed for object-type schemas.
+     * 
+     * <p>
+     * Implementations MAY ignore this property if this schema's {@linkplain #type() type} is not
+     * {@linkplain SchemaType#OBJECT OBJECT}, either explicitly or as derived by the placement of the annotation.
+     * 
+     * @return a class that describes the allowable schema for additional properties not explicitly defined
+     * 
+     * @since 3.1
+     * 
+     * @see org.eclipse.microprofile.openapi.models.media.Schema.TrueSchema
+     * @see org.eclipse.microprofile.openapi.models.media.Schema.FalseSchema
+     */
+    Class<?> additionalProperties() default org.eclipse.microprofile.openapi.models.media.Schema.TrueSchema.class;
```

Comment:
It seems odd that `properties` is a `SchemaProperty` while `additionalProperties` is a `Class<?>` when they seem to document very similar things.

Reply:
Just to add, most of the other attribute that specify a "remote" schema utilize the `Class<?>` approach. `properties` is a special case to allow the fields of an object-type to be explicitly listed out in-line, allowing overrides.

https://github.com/eclipse/microprofile-open-api/blob/2f6592e609489c7dc47eb8698f63d86d7e8ef27c/tck/src/main/java/org/eclipse/microprofile/openapi/apps/airlines/JAXRSApp.java#L78-L79

Reply:
Thought about it a bit more and I think we should leave it as `Class<?>` like the other attributes. I've written a bit more in my comment below.